src/FRP/Event/Time.purs

```diff
@@ -15,3 +17,38 @@ foreign import animationFrame :: Event Unit
 
 -- | Create an event which reports the current time in milliseconds since the epoch.
 foreign import withTime :: forall a. Event a -> Event { value :: a, time :: Number }
+
+-- | Provided an input event and transformation, block the input event for the
+-- | duration of the specified period on each output.
+withBlocking
+  :: forall a b.
+     (Event a -> Event { period :: Number, value :: b })
+  -> Event a
+  -> Event b
+withBlocking process event
+  = fix \allowed ->
+      let
+        processed :: Event { period :: Number, value :: b }
+        processed = process allowed
+
+        expiries :: Event Number
+        expiries = pure 0.0 <|>
```

Comment:
Zero seems like an odd choice here. What about representing the expiry as a predicate on times instead? Then you could start with `const true` here.

Reply:
Am I then saying that each loop adds a new predicate on whether times are above its expiry? Is there a perf concern here? (I'll do some profiling over long time periods)

Reply:
Yeah, I agree, 0.0 isn't the best ... What about using `Maybe` instead?

Reply:
Sounds good.